conference/react-js/src/components/Header/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { PureComponent } from 'react';
+import SignInForm from '../../containers/SignInForm';
+
+class Header extends PureComponent {
+  constructor() {
+    super();
+
+    this.onButtonAccountClick = this.onButtonAccountClick.bind(this);
+    this.state = { visible: true };
+  }
+
+  onButtonAccountClick() {
+    const currentState = this.state.visible;
```

Comment:
When You using setState based on the previous state use setState with callback parameter instead of an object. 

```
this.setState(({ visible }) => ({ visible: !visible }))
```

https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b